src/apps/explorer/components/OrdersTableWidget/useGetOrders.tsx

```diff
@@ -20,24 +20,34 @@ type Result = {
   orders: Order[]
   error: string
   isLoading: boolean
+  isThereNext: boolean
 }
 
-export function useGetOrders(ownerAddress: string, limit = 1000, offset = 0): Result {
+export function useGetOrders(ownerAddress: string, limit = 1000, offset = 0, pageIndex?: number): Result {
   const [isLoading, setIsLoading] = useState(false)
   const [error, setError] = useState('')
   const [orders, setOrders] = useState<Order[]>([])
   const networkId = useNetworkId() || undefined
   const [erc20Addresses, setErc20Addresses] = useState<string[]>([])
   const { value: valueErc20s, isLoading: areErc20Loading } = useMultipleErc20({ networkId, addresses: erc20Addresses })
   const [mountNewOrders, setMountNewOrders] = useState(false)
+  const [isThereNext, setIsThereNext] = useState(false)
 
   const fetchOrders = useCallback(
     async (network: Network, owner: string): Promise<void> => {
       setIsLoading(true)
       setError('')
+      const limitPlusOne = limit + 1
+      const checkForNext = (_orders: RawOrder[]): void => {
+        if (_orders.length === limitPlusOne) {
+          setIsThereNext(true)
+          _orders.pop()
+        }
+      }
 
       try {
-        const ordersFetched = await getAccountOrders({ networkId: network, owner, offset, limit })
+        const ordersFetched = await getAccountOrders({ networkId: network, owner, offset, limit: limitPlusOne })
+        checkForNext(ordersFetched)
```

Comment:
Since the function is instantiated at every run, I don't see much benefit on having it.

I would either move it outside in another callback or just do the `if` right here

Reply:
Good point.